### PR TITLE
Adds nwc requester

### PR DESCRIPTION
**Pull request by shreyav**

### CODE REVIEW

**Comment by shreyav** [comment]
* **#127** <a href="https://app.graphite.dev/github/pr/uma-universal-money-address/uma-js-sdk/127?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @shreyav and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/uma-auth-client/src/Nip47Types.ts [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+export type LookupInvoiceRequest = {
+  payment_hash: string;
+};
+
+// Note: when we have multiple receiver types like bolt12 etc, these will be optional and it is expected that exactly one of the fields is set.
+export type Receiver = {
+  lud16: string;
+};
+
+export type LookupUserRequest = {
```
Comment (shreyav) [inline packages/uma-auth-client/src/Nip47Types.ts:116]: changed from:
```
export type Nip47LookupUserRequest = {
  lud16: string;
};
```

File: packages/uma-auth-client/src/Nip47Types.ts [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+export type LookupUserResponse = {
+  receiver: Receiver;
+  currencies: Currency[];
+};
+
+export type FetchQuoteRequest = {
+  receiver: Receiver;
+  sending_currency_code: string;
+  receiving_currency_code: string;
+  locked_currency_side: "SENDING" | "RECEIVING";
```
Comment (shreyav) [inline packages/uma-auth-client/src/Nip47Types.ts:130]: does case matter here? if so I probably need to change it on android

File: packages/uma-auth-client/src/Nip47Types.ts [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+export type PayToAddressRequest = {
+  receiver: Receiver;
+  sending_currency_code: string;
+  sending_currency_amount: number;
+  receiving_currency_code?: string;
+};
+
+export type PayToAddressResponse = {
+  preimage: string;
+  quote: Quote;
```
Comment (shreyav) [inline packages/uma-auth-client/src/Nip47Types.ts:162]: added quote here

File: packages/uma-auth-client/src/NwcRequester.ts [hunk 4]
```diff
@@ -0,0 +1,416 @@
+import {
```
Comment (jklein24) [inline packages/uma-auth-client/src/NwcRequester.ts:2]: Might be good to include a comment at the top of this file indicating it was adapted from the alby sdk with a link to that file for reference (and to give them credit :-))

**Review by jklein24 — APPROVED** [review]
Awesome!

File: packages/uma-auth-client/src/NwcRequester.ts (hunk 4 above)
Comment (shreyav) [reply packages/uma-auth-client/src/NwcRequester.ts:2]: ooh good idea!! ty

**Pull request merged**